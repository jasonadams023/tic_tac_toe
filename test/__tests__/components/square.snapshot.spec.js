import React from 'react';
import Square from '../../../app/components/square.js';

describe('Square', () => {
    it('is empty when value is null', () => {
        const wrapper = mount(
            <Square
                key={ '0' + '-' + '0' }
                value={ null }
                onClick={ () => this.props.onClick('0', '0') }
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('displays X when value is X', () => {
        const wrapper = mount(
            <Square
                key={ '0' + '-' + '0' }
                value={ 'X' }
                onClick={ () => this.props.onClick('0', '0') }
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});