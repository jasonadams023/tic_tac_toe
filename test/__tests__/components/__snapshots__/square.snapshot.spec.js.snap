// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square displays X when value is X 1`] = `
<Square
  onClick={[Function]}
  value="X"
>
  <button
    className="square"
    onClick={[Function]}
  >
    X
  </button>
</Square>
`;

exports[`Square is empty when value is null 1`] = `
<Square
  onClick={[Function]}
  value={null}
>
  <button
    className="square"
    onClick={[Function]}
  />
</Square>
`;
